doc/userguide/firewall/firewall-example.rst

```diff
@@ -0,0 +1,93 @@
+Firewall Ruleset Examples
+=========================
+
+.. note:: In Suricata 8 the firewall mode is experimental and subject to change.
+
+HTTP
+----
+
+In this example a simple HTTP ruleset will be shown. It will allow HTTP to flow
+as long as:
+
+- method is GET or POST
+- User-Agent is "curl"
+- Status code is 200.
+
+It starts by allowing the TCP port 80 traffic.
+
+::
+
+    accept:hook tcp:all any any <> any 80 (sid:10;)
+
+The stream tracking combined with the default exception policy handling will enforce
+a proper TCP handshake, etc.
+
+The HTTP rules need to ``accept`` each state::
+
+    # allow traffic before the request line is complete
+    accept:hook http1:request_not_started any any -> any any (sid:100;)
+    # allow GET
+    accept:hook http1:request_line any any -> any any ( \
+            http.method; content:"GET"; sid:101;)
+    # or allow POST
+    accept:hook http1:request_line any any -> any any ( \
+            http.method; content:"POST"; sid:102;)
```

Comment:
Later, we can probably use the rule example containers here to highlight specific parts.